DBZ-6090 Added documentation for mongodb.connection.mode (#4401)
debezium/debezium: documentation/modules/ROOT/pages/connectors/mongodb.adoc

 +
[NOTE]
====
Connectors that capture changes from a sharded MongoDB cluster use this connection string only during the initial shard discovery process when xref:mmongodb-property-mongodb-connection-mode[`mongodb.connection.mode`] is set to `replica_set`.
After the initial discovery process, connection strings are generated for each individual shard.
====

|[[mongodb-property-mongodb-connection-mode]]<<mongodb-property-mongodb-connection-mode, `+mongodb.connection.mode+`>>
|`replica_set`
|Specifies the strategy used by the connector when connecting to `sharded` MongoDB cluster.
Set this property to one of the following values:

> `replica_set`:: individually connect to each replica set / shard .

Author: @roldanbob the `or` is a mistake here. Sharded cluster is composed of shard and each shard is a replica set.

> `replica_set`:: individually connect to each replica set / shard .

Reviewer: I updated the description based on that understanding. Let me know if the new version works.

> `replica_set`:: individually connect to each replica set / shard .

Author: This wording is correct.

> @jcechace So, looking at the diagram [here](https://www.mongodb.com/docs/manual/core/sharded-cluster-components/#production-configuration), would the connector attach to the primary node in each shard's replica set?

This is tricky... the exact node the client is communicating with can wary based on configuration (e.g. ReadPreference). For example the client will route write operations always to the primary but can route read operations to one of the secondaries if available (this is actually the configuration used by the connector now). So it's better to think about it as connecting to the Replica Set as a whole rather than connecting to any specific node within it.

|[[mongodb-property-tasks-max]]<<mongodb-property-tasks-max, `+tasks.max+`>>
|`1`

> |The maximum number of tasks that should be created for this connector. The MongoDB connector will attempt to use a separate task for each replica set. We recommend specifying a value that is equal to or more than the number of shards in the cluster, so that the work for each replica set can be distributed by Kafka Connect.

Author: Yes, this indeed applies only when connected to a sharded cluster. 

However this part is not right `contains multiple sharded replica sets`  -- a replica set is not sharded, a server is and each shard is a replica set.

> |The maximum number of tasks that should be created for this connector. The MongoDB connector will attempt to use a separate task for each replica set. We recommend specifying a value that is equal to or more than the number of shards in the cluster, so that the work for each replica set can be distributed by Kafka Connect.

Reviewer: @jcechace I updated the description. Let me know if this new version works.

> |The maximum number of tasks that should be created for this connector. The MongoDB connector will attempt to use a separate task for each replica set. We recommend specifying a value that is equal to or more than the number of shards in the cluster, so that the work for each replica set can be distributed by Kafka Connect.

Author: LGTM except there is an extra space after comma in `But when a cluster contains multiple shards,  to enable`

[NOTE]
====
This property has an effect only when the connector is connected to a sharded MongoDB cluster and the xref:mongodb-property-mongodb-connection-mode[`mongodb.connection.mode`] property is set to `replica_set`.

> When xref:mongodb-property-mongodb-connection-mode[`mongodb.connection.mode`] is set to `sharded` or the connector is connected to a replica set deployment of MongoDB then only a single task will be   used.

Reviewer: ```suggestion
IThe connector ignores this settings and uses only a single task if xref:mongodb-property-mongodb-connection-mode[`mongodb.connection.mode`] is set to `sharded`, or if the connector is connected to an unsharded MongoDB replica set deployment.  
```
